node/src/components/consensus/era_supervisor/tests.rs

```diff
@@ -0,0 +1,120 @@
+use anyhow::Error;
+use casper_execution_engine::{core::engine_state::genesis::GenesisAccount, shared::motes::Motes};
+use derive_more::From;
+use prometheus::Registry;
+
+use super::*;
+use crate::{
+    components::{
+        consensus::{
+            tests::mock_proto::{Message, MockProto, NodeId},
+            Config,
+        },
+        storage::Storage,
+        Component,
+    },
+    crypto::asymmetric_key::{PublicKey, SecretKey},
+    effect::{
+        announcements::ConsensusAnnouncement,
+        requests::{
+            BlockExecutorRequest, BlockProposerRequest, BlockValidationRequest,
+            ContractRuntimeRequest, NetworkRequest, StorageRequest,
+        },
+        EffectBuilder,
+    },
+    protocol,
+    reactor::{EventQueueHandle, QueueKind, Scheduler},
+    testing::TestRng,
+    utils::{self, External, Loadable},
+};
+
+type ClMessage = Message<ClContext>;
+
+#[derive(Debug, From)]
+enum Event {
+    #[from]
+    Consensus(super::Event<NodeId>),
+    #[from]
+    Network(NetworkRequest<NodeId, protocol::Message>),
+    #[from]
+    BlockProposer(BlockProposerRequest),
+    #[from]
+    ConsensusAnnouncement(ConsensusAnnouncement),
+    #[from]
+    BlockExecutor(BlockExecutorRequest),
+    #[from]
+    BlockValidator(BlockValidationRequest<ProtoBlock, NodeId>),
+    #[from]
+    Storage(StorageRequest<Storage>),
+    #[from]
+    ContractRuntime(ContractRuntimeRequest),
+}
+
+/// Loads the local chainspec and overrides timestamp and genesis account with the given stakes.
+fn new_test_chainspec(stakes: Vec<(PublicKey, u64)>) -> Chainspec {
+    let mut chainspec = Chainspec::from_resources("local/chainspec.toml");
+    chainspec.genesis.accounts = stakes
+        .into_iter()
+        .map(|(pk, stake)| {
+            let motes = Motes::new(stake.into());
+            GenesisAccount::new(pk.into(), pk.to_account_hash(), motes, motes)
+        })
+        .collect();
+    chainspec.genesis.timestamp = Timestamp::now() + 1000.into();
+    chainspec.genesis.highway_config.genesis_era_start_timestamp = chainspec.genesis.timestamp;
+    chainspec
+}
+
+#[test]
+fn era_supervisor() -> Result<(), Error> {
+    let mut rng = TestRng::new();
+
+    let alice_sk = SecretKey::random(&mut rng);
+    let alice_pk = PublicKey::from(&alice_sk);
+    let bob_sk = SecretKey::random(&mut rng);
+    let bob_pk = PublicKey::from(&bob_sk);
+
+    let chainspec = new_test_chainspec(vec![(alice_pk, 100)]);
+    let config = Config {
+        secret_key_path: External::Loaded(alice_sk),
+    };
+
+    let registry = Registry::new();
+    let scheduler = utils::leak(Scheduler::<Event>::new(QueueKind::weights()));
+    let event_queue = EventQueueHandle::new(&scheduler);
+    let effect_builder = EffectBuilder::new(event_queue);
```

Comment:
Maybe:

```rust
let effect_builder = {
   let scheduler = utils::leak(Scheduler::<Event>::new(QueueKind::weights()));
   let event_queue = EventQueueHandle::new(&scheduler);
   EffectBuilder::new(event_queue)
}
```

?

Reply:
…or maybe even another helper function?

Reply:
Sure, a helper function would work here.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/502/commits/c09bbb61ce7f53c2c4d56d0731ea14495300702b.